Selecting the wallpaper won't error, but instead select the entire screen
Made by https://github.com/vizs
diff --git a/crud.c b/crud.c
index b9cdf8f..f88a62d 100644
--- a/crud.c
+++ b/crud.c
@@ -228,15 +228,25 @@ int main(int argc, char **argv) {
         unsigned int m;
 
         if (XQueryPointer(display, root, &hroot, &hovered, &rx, &ry, &hx, &hy, &m)) {
-          XWindowAttributes attrs;
-          XGetWindowAttributes(display, hovered, &attrs);
-          // update with the selected window's attributes
-          set_selection(&ws, (Selection) {
-            .x = attrs.x,
-            .y = attrs.y,
-            .width = attrs.width,
-            .height = attrs.height
-          });
+          if (hovered ==  0) {
+            Screen* sscreen = ScreenOfDisplay(display, DefaultScreen(display));
+            set_selection(&ws, (Selection) {
+              .x = 0,
+              .y = 0,
+              .width = WidthOfScreen(sscreen),
+              .height = HeightOfScreen(sscreen)
+            });
+          } else {
+            XWindowAttributes attrs;
+            XGetWindowAttributes(display, hovered, &attrs);
+            // update with the selected window's attributes
+            set_selection(&ws, (Selection) {
+              .x = attrs.x,
+              .y = attrs.y,
+              .width = attrs.width,
+              .height = attrs.height
+            });
+          }
         }
       }
 
